sklearn/ensemble/_hist_gradient_boosting/histogram.pyx

```diff
@@ -234,21 +259,33 @@ cdef class HistogramBuilder:
 
         cdef:
             int feature_idx
+            int f_idx
             int n_features = self.n_features
+            int n_allowed_features = self.n_features
             hist_struct [:, ::1] histograms = np.empty(
                 shape=(self.n_features, self.n_bins),
                 dtype=HISTOGRAM_DTYPE
             )
+            bint has_interaction_cst = False
             int n_threads = self.n_threads
 
-        for feature_idx in prange(n_features, schedule='static', nogil=True,
-                                  num_threads=n_threads):
-            # Compute histogram of each feature
+        has_interaction_cst = allowed_features is not None
+        if has_interaction_cst:
+            n_allowed_features = allowed_features.shape[0]
+
+        # Compute histogram of each feature
+        for f_idx in prange(n_allowed_features, schedule='static', nogil=True,
+                            num_threads=n_threads):
+            if has_interaction_cst:
+                feature_idx = allowed_features[f_idx]
+            else:
+                feature_idx = f_idx
+
             _subtract_histograms(feature_idx,
-                                 self.n_bins,
-                                 parent_histograms,
-                                 sibling_histograms,
-                                 histograms)
+                                self.n_bins,
+                                parent_histograms,
+                                sibling_histograms,
+                                histograms)
```

Comment:
Nit. I can't suggest, but I guess this can be reformated _à la_ `black`? 
```python
            _subtract_histograms(
                feature_idx,
                self.n_bins,
                parent_histograms,
                sibling_histograms,
                histograms,
            )
```